metric/syncint64.go

```diff
@@ -147,8 +147,9 @@ type Int64Histogram interface {
 // Int64HistogramConfig contains options for synchronous counter instruments
 // that record int64 values.
 type Int64HistogramConfig struct {
-	description string
-	unit        string
+	description              string
+	unit                     string
+	explicitBucketBoundaries []float64
```

Comment:
I am concerned about potential `float64` -> `int64` rounding errors 😬 

The `WithFloat64HistogramBoundaries` and `WithInt64HistogramBoundaries` proposal looks safer and should avoid such errors.

Reply:
In both cases, the [boundaries are float64](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-advisory-parameter-explicitbucketboundaries).  The int64 refers to the kind of observations accepted by the instrument.  As is the case today, there can be rounding errors when comparing integer observations with float64 boundaries.

This is just a question of how to split the options

Reply:
For reference: https://go.dev/play/p/QUv8t-zbIez

Reply:
If the boundaries HAVE to be `float64`  then I am totally for the current proposal.

However, I am not sure if this is not a bug in the specification. I think it is the only place which "requires" a concrete type. Maybe it is just a recommendation? Maybe it makes sense to discuss this in the specification? @jack-berg What is your opinion?

Reply:
As a note that this matches the configuration available via the View API as well: https://github.com/open-telemetry/opentelemetry-go/blob/3f17bcb6c309157ec256be9197149014efea1c42/sdk/metric/aggregation.go#L108

The specification also requires bucket boundaries on view configuration to be floats/doubles: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#explicit-bucket-histogram-aggregation

Reply:
> As a note that this matches the configuration available via the View API as well:

This is a very good argument for me 😉

Reply:
In java, we have both int64 and float64 variants of histogram instruments. The APIs for each of these accept the corresponding types. I.e. you record int64 measurements to the int64 histogram, and float64 measurements to the float64 histogram. We extended this to the APIs for setting the bucket boundaries advice. Under the covers, we translate the int64 bucket boundaries to float64 because as @dashpole points out, that's how we ultimately need to serialize the data. So accepting int64 for int64 histograms is really just syntactic sugar, and not a precision thing, since there's no way to retain the int64 precision when serializing.